src/main/java/bio/terra/service/filedata/google/gcs/GcsPdao.java

```diff
@@ -380,19 +381,19 @@ public void deleteFile(FireStoreFile fireStoreFile) {
     if (fireStoreFile != null) {
       GoogleBucketResource bucketResource =
           resourceService.lookupBucket(fireStoreFile.getBucketResourceId());
-      deleteFileByGspath(fireStoreFile.getGspath(), bucketResource);
+      deleteFileByGspath(fireStoreFile.getGspath(), bucketResource.projectIdForBucket());
     }
   }
 
-  private boolean deleteWorker(GoogleBucketResource bucketResource, String bucketPath) {
-    GcsProject gcsProject =
-        gcsProjectFactory.get(bucketResource.getProjectResource().getGoogleProjectId());
+  private boolean deleteWorker(BlobId blobId, String projectId) {
+    GcsProject gcsProject = gcsProjectFactory.get(projectId);
     Storage storage = gcsProject.getStorage();
-    Blob blob = storage.get(BlobId.of(bucketResource.getName(), bucketPath));
+    Blob blob = storage.get(blobId);
     if (blob != null) {
       return blob.delete();
     }
-    logger.warn("{} was not found and so deletion was skipped", bucketPath);
+    logger.warn(
+        "{} was not found and so deletion was skipped", GcsUriUtils.getGsPathFromBlob(blobId));
```

Comment:
IMO you could use the `blobId` directly here, as this message will only be seen by us anyway, so it doesn't need to formatted as a `gs://` path.

Reply:
Oh true. Plus, that makes it fit on one line 😉